draft-ietf-core-oscore-groupcomm.md

```diff
@@ -1020,7 +1031,47 @@ When receiving a request from a client for the first time, the server is not syn
+
+* For requests, notifications, and non-notification responses to group requests, Group OSCORE also provides relative freshness. That is, the received Partial IV allows a recipient to determine the relative order of requests or responses.
+
+* It is not guaranteed that a misbehaving server did not create the response before receiving the request, i.e., Group OSCORE does not verify server aliveness.
+
+## Replay Protection # {#sec-replay-protection}
+
+Like in OSCORE {{RFC8613}}, the replay protection relies on the Partial IV of incoming messages. The operation of validating the Partial IV and performing replay protection MUST be atomic.
+
+The protection from replay of requests is performed as per {{Section 7.4 of RFC8613}}, separately for each client by leveraging the Replay Window in the corresponding Recipient Context. When supporting Observe {{RFC7641}}, the protection from replay of notifications is performed as per {{Section 7.4.1 of RFC8613}}.
+
+### Replay Protection of Non-notification Responses# {#sec-replay-protection-non-notifications}
+
+This section refers specifically to non-notification responses to a group request. A client can receive multiple such responses from the same server in the group as a reply to the same group request, until the CoAP Token value associated with the group request is freed up {{I-D.ietf-core-groupcomm-bis}}.
+
+When replying to a group request with a non-notification response (both successful and error), a server MUST include a Partial IV, except for the first non-notification response where the Partial IV MAY be omitted. A server supporting Observe {{RFC7641}} MUST NOT reply to a group request with 2.xx responses of which some are notifications and some are not.
+
+When processing responses from a same server to an Observe registration request, a client supporting Observe MUST accept either only notifications or only non-notification responses. The specific way to achieve this is implementation specific.
+
+In order to protect against replay, the client SHALL maintain for each ongoing Non-Notification Group Exchange one Response Number for each different server. The Response Number is a non-negative integer containing the largest Partial IV of the received non-notification responses from that server within the Non-Notification Group Exchange.
+
+Then, separately for each server, the client uses the associated Response Number to perform ordering and replay protection of the non-notification responses to a group request received from that server, by comparing their Partial IVs with one another and against the Response Number.
+
+For each server, the associated Response Number is initialized to the Partial IV of the first successfully verified non-notification response to a group request. A client MUST only accept at most one such response without Partial IV from each server in the group, and treat it as the oldest non-notification response to the group request received from that server.
```

Comment:
_the associated Response Number is initialized to the Partial IV of the first successfully verified non-notification response_ Meaning the first one that actually contains a Partial IV? (As the first one may not).

Reply:
Yes, however:

- This has to be read together with the later sentence: " Given a group request, implementations need to make sure that the corresponding non-notification response from a server without Partial IV is considered the oldest from that server."

- This text has the same construction and uses the same parlance as the text in Section 4.1.3.5.2 of RFC 8613 about how OSCORE handles notifications.

It's good to mirror as much as possible what OSCORE says for notifications.